scripts/__tgm_core/__tgm_core.gml

```diff
@@ -1680,43 +1680,37 @@ function gui_to_room_dimension_ext(x1, y1, camera, angle, gui_width, gui_height,
 
 
 /// @desc Transforms a 2D coordinate (in window space) to a 3D vector.
-/// Returns an array of the following format:
-/// [dx, dy, dz, ox, oy, oz]
-/// where [dx, dy, dz] is the direction vector and [ox, oy, oz] is the origin of the ray.
+/// Returns a Vector3 containing a vector representing the mouse's vector cast into a 3D scene.
```

Comment:
Depending on what will be done in the screen_to_world_dimension function, I believe that a Vector2 is enough (x and y), but if you want to return a Vector3 with something else, let me know.

Reply:
screen_to_world is usually in 3D to let the user cast a ray from the position of the camera, to do things like throwing a projectile or screen picking something under the cursor or something. You can't do that with a Vector2.